pkg/otlp/model/translator/metrics_translator.go

```diff
@@ -153,6 +160,11 @@ func (t *Translator) mapNumberMonotonicMetrics(
 			continue
 		}
 
+		if i == 0 && getProcessStartTime() < startTs {
+			// Report the first value if the timeseries started after the Datadog Agent process started.
+			consumer.ConsumeTimeSeries(ctx, pointDims, Count, ts, val)
+		}
+
 		if dx, ok := t.prevPts.MonotonicDiff(pointDims, startTs, ts, val); ok {
 			consumer.ConsumeTimeSeries(ctx, pointDims, Count, ts, dx)
 		}
```

Comment:
The way I was thinking about it was this:

```suggestion
		if dx, ok := t.prevPts.MonotonicDiff(pointDims, startTs, ts, val); ok {
			consumer.ConsumeTimeSeries(ctx, pointDims, Count, ts, dx)
		} else if i == 0 && getProcessStartTime() < startTs {
			// Report the first value if the timeseries started after the Datadog Agent process started.
			consumer.ConsumeTimeSeries(ctx, pointDims, Count, ts, val)
		}
```

since the first point may have come on a previous payload. In general, I think we should never record `val` if we have something in cache related to a given timeseries.

Does that make sense?

Reply:
Makes sense to me, this way it also makes the (exclusive) conditions clearer. Updated.